vendor/bat-native-ads/src/bat/ads/internal/ml/model/linear/linear_unittest.cc

```diff
@@ -0,0 +1,143 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <cmath>
+#include <vector>
+
+#include "bat/ads/internal/ml/data/vector_data.h"
+#include "bat/ads/internal/ml/model/linear/linear.h"
+
+#include "bat/ads/internal/json_helper.h"
+#include "bat/ads/internal/unittest_base.h"
+#include "bat/ads/internal/unittest_util.h"
+
+// npm run test -- brave_unit_tests --filter=BatAds*
+
+namespace ads {
+namespace ml {
+
+class BatAdsLinearModelTest : public UnitTestBase {
+ protected:
+  BatAdsLinearModelTest() = default;
+
+  ~BatAdsLinearModelTest() override = default;
+};
+
+TEST_F(BatAdsLinearModelTest, ThreeClassesPredictionTest) {
+  // Arrange
+  const std::map<std::string, VectorData> weights = {
+      {"class_1", VectorData(std::vector<double>{1.0, 0.0, 0.0})},
+      {"class_2", VectorData(std::vector<double>{0.0, 1.0, 0.0})},
+      {"class_3", VectorData(std::vector<double>{0.0, 0.0, 1.0})}};
+
+  const std::map<std::string, double> biases = {
+      {"class_1", 0.0}, {"class_2", 0.0}, {"class_3", 0.0}};
+
+  const model::Linear linear(weights, biases);
+  const VectorData class1_data_vector(std::vector<double>{1.0, 0.0, 0.0});
```

Comment:
nit: object is known as VectorData but variable is named `data_vector`

Reply:
Done.